applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.h

```diff
@@ -0,0 +1,320 @@
+{
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+template <int Dim, int NumNodes>
+class TransonicPerturbationPotentialFlowElement : public Element
+{
+public:
+    template <unsigned int TNumNodes, unsigned int TDim>
+    struct ElementalData
+    {
+        array_1d<double, TNumNodes> potentials, distances;
+        double vol;
+
+        BoundedMatrix<double, TNumNodes, TDim> DN_DX;
+        array_1d<double, TNumNodes> N;
+    };
+
+    ///@name Type Definitions
+    ///@{
+
+    typedef Element BaseType;
+    static constexpr int TNumNodes = NumNodes;
+    static constexpr int TDim = Dim;
+
+    ///@}
+    ///@name Pointer Definitions
+    /// Pointer definition of TransonicPerturbationPotentialFlowElement
+    KRATOS_CLASS_POINTER_DEFINITION(TransonicPerturbationPotentialFlowElement);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    // Constructors.
+
+    /// Default constuctor.
+    /**
+     * @param NewId Index number of the new element (optional)
+     */
+    explicit TransonicPerturbationPotentialFlowElement(IndexType NewId = 0)
+    {
+    }
+
+    /**
+     * Constructor using an array of nodes
+     */
+    TransonicPerturbationPotentialFlowElement(IndexType NewId, const NodesArrayType& ThisNodes)
+        : Element(NewId, ThisNodes)
+    {
+    }
+
+    /**
+     * Constructor using Geometry
+     */
+    TransonicPerturbationPotentialFlowElement(IndexType NewId, GeometryType::Pointer pGeometry)
+        : Element(NewId, pGeometry)
+    {
+    }
+
+    /**
+     * Constructor using Properties
+     */
+    TransonicPerturbationPotentialFlowElement(IndexType NewId,
+                                     GeometryType::Pointer pGeometry,
+                                     PropertiesType::Pointer pProperties)
+        : Element(NewId, pGeometry, pProperties)
+    {
+    }
+
+    /**
+     * Copy Constructor
+     */
+    TransonicPerturbationPotentialFlowElement(TransonicPerturbationPotentialFlowElement const& rOther) = delete;
+
+    /**
+     * Move Constructor
+     */
+    TransonicPerturbationPotentialFlowElement(TransonicPerturbationPotentialFlowElement&& rOther) = delete;
+
+    /**
+     * Destructor
+     */
+    ~TransonicPerturbationPotentialFlowElement() override
+    {
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /// Assignment operator.
+    TransonicPerturbationPotentialFlowElement& operator=(TransonicPerturbationPotentialFlowElement const& rOther) = delete;
+
+    /// Move operator.
+    TransonicPerturbationPotentialFlowElement& operator=(TransonicPerturbationPotentialFlowElement&& rOther) = delete;
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    Element::Pointer Create(IndexType NewId,
+                            NodesArrayType const& ThisNodes,
+                            PropertiesType::Pointer pProperties) const override;
```

Comment:
```suggestion
    Element::Pointer Create(
        IndexType NewId,
        NodesArrayType const& ThisNodes,
        PropertiesType::Pointer pProperties) const override;
```
This is the format style we normally use (same for the other methods).